src/ansys/dpf/post/selection.py

```diff
@@ -0,0 +1,557 @@
+        """
+        scoping = self._evaluate_on(simulation=simulation)
+        return scoping.ids
+
+
+class SpatialSelection:
+    """Define a spatial selection.
+
+    Translate space selection types (node ids, elements ids, geometry selections,
+    named selections...) into DPF known entities.
+    """
+
+    def __init__(
+        self,
+        scoping: Union[Scoping, None] = None,
+        server: Union[BaseServer, None] = None,
+    ):
+        """Create a SpatialSelection.
+
+        Parameters
+        ----------
+        scoping:
+            Scoping to use to define the SpatialSelection.
+        server:
+            DPF server on which to create the SpatialSelection.
+        """
+        self._server = get_or_create_server(server)
+        self._selection = Workflow(server=self._server)
+
+        if scoping is not None:
+            self.select_with_scoping(scoping)
+
+    def select_named_selection(
+        self,
+        named_selection: Union[str, List[str]],
+        location: Union[str, locations, None] = None,
+    ) -> None:
+        """Select a mesh scoping corresponding to one or several named selections.
+
+        Parameters
+        ----------
+        named_selection:
+            Name of the named selection.
+
+        location:
+            Location to use (nodal, elemental...)
+        """
+        if isinstance(named_selection, str):
+            op = operators.scoping.on_named_selection(
+                requested_location=location,
+                named_selection_name=named_selection,
+                server=self._server,
+            )
+            self._selection.set_input_name(
+                _WfNames.data_sources, op.inputs.data_sources
+            )
+            self._selection.set_input_name(
+                _WfNames.streams, op.inputs.streams_container
+            )
+            self._selection.set_output_name(_WfNames.scoping, op.outputs.mesh_scoping)
+        else:
+            op = operators.utility.merge_scopings(server=self._server)
+            forward_ds = operators.utility.forward(any=None, server=self._server)
+            forward_sc = operators.utility.forward(any=None, server=self._server)
+            self._selection.set_input_name(_WfNames.data_sources, forward_ds.inputs.any)
+            self._selection.set_input_name(_WfNames.streams, forward_sc.inputs.any)
+            for pin, ns in enumerate(named_selection):
+                mesh_scoping_op = operators.scoping.on_named_selection(
+                    requested_location=location,
+                    named_selection_name=ns,
+                    server=self._server,
+                    # data_sources=forward_ds.outputs.any,
+                    # streams_container=forward_sc.outputs.any,
+                )
+                mesh_scoping_op.connect(3, forward_sc.outputs.any)
+                mesh_scoping_op.connect(4, forward_ds.outputs.any)
+                op.connect(pin, mesh_scoping_op.outputs.mesh_scoping)
```

Comment:
I'm not sure what is happening here...

Reply:
This code treats the case of several named_selections being selected. It instantiates a scoping merging operator, then for each named selection connects the scoping obtained with scoping_on_named_selection.